frappe/email/doctype/newsletter/newsletter.js

```diff
@@ -122,6 +122,15 @@ frappe.ui.form.on("Newsletter", {
 			],
 			primary_action_label: __("Schedule"),
 			primary_action({ date, time }) {
+				if (typeof date === "undefined") {
+					frappe.throw(__("Please provide date for Scheduling."));
+				}
+
+				const today = new Date().toJSON().slice(0, 10);
+
+				if (date < today) {
+					frappe.throw(__("Past dates are not allowed for Scheduling."));
+				}
```

Comment:
As a general rule, all business logic should be in server-side code. There can be more than one client (e.g. APIs) who wont benefit from this validation.